partner/WHMCS/modules/servers/jetpack/jetpack.php

```diff
@@ -0,0 +1,288 @@
+
+    if ($response && $http_code >= 400) {
+        if ($data['client_id']) {
+            //If client_id is set in the data array when this function was called then we are trying to get an access
+            // token and are unsuccessful which likely means the client id or client secret are not valid.
+            return 'JETPACK MODULE: There was a problem getting an access token for your Jetpack hosting partner 
+            account. This usually means the Client Id or Client Secret provided when creating the product are invalid';
+        } elseif ($data['siteurl']) {
+            //If the siteurl is set in the data array we are attempting to provision a plan. There are few errors that
+            //could show up here but its likely the user did not enter siteurl or local_user field correctly.
+            return 'JETPACK MODULE: There was a problem provisioning a Jetpack Plan for ' . $data['siteurl'] . '. The 
+            response from the request was ' . $response . '. This usually means the url provided does not have a valid 
+            WordPress installation or the local_user field provided for provisioning is not valid.';
+        }
+        return 'JETPACK MODULE: There was a problem with provisioning or terminating a Jetpack Plan. 
+        The response was ' . $response;
+    } elseif (curl_error($curl)) {
+        throw new Exception('Unable to connect: ' . curl_errno($curl) . ' - ' . curl_error($curl));
+    } elseif (empty($response)) {
+        throw new Exception('Empty response');
+    }
+
+    curl_close($curl);
+    return json_decode($response);
+}
+
+/**
+ * Save the next_url for Jetpack activation/setup to the
+ * order for the client
+ *
+ * @param $url
+ * @param $orderId
+ */
+function save_provisioning_details($url, $params, $pending = false)
+{
+    $jetpack_next_url_field = Capsule::table('tblcustomfields')
+        ->where(array('fieldname' => 'jetpack_provisioning_details', 'type' => 'product'))->first();
+
+    $details = '';
+    if ($url) {
+        $details = 'URL to Activate Jetpack: ' . $url;
+    } elseif ($pending) {
+        $details = 'The domain did not appear to resolve when provisioning was attempted however a Jetpack plan is 
+        waiting for ' . $params['customfields']['Site URL'] . '. Once DNS resolves please connect the site via 
+        the Jetpack Banner in the sites dashboard';
+    }
+    Capsule::table('tblcustomfieldsvalues')->where(array('fieldid' => $jetpack_next_url_field->id))->update(array(
+        'relid' => $params['model']['orderId'], 'value' => $details));
+}
+
+/**
+ * Validate that the module was correctly set up when the product was
+ * created by the WHMCS user and that the required Fields/Options for
+ * being able to provision a Jetpack plan are present. Fields validated are
+ *  - Allowed Plans from Plan Custom Field
+ *  - Required Custom Fields
+ *  - Required Config Options
+ *
+ * @param array $params
+ * @return bool
```

Comment:
When there is an error it returns a string.

Reply:
@nb I've updated this to reflect the mixed return type and also updated the functions that call this to check for the boolean value or return the error string.